pkgs/build-support/rust/mk-cargo-vendor-deps-script.py

```diff
@@ -0,0 +1,223 @@
+
+    if len(registry_packages) != 0:
+        os.mkdir(os.path.join(out_dir, "registry"))
+
+        # run download jobs in parallel
+        pool = mp.Pool(min(10, mp.cpu_count()))
+        pool.map(download_tarball, ((pkg, out_dir) for pkg in registry_packages))
+        pool.close()
+
+
+def get_manifest_metadata(manifest_path: str):
+    output = subprocess.check_output(["cargo", "metadata", "--format-version", "1", "--no-deps", "--manifest-path", manifest_path])
+    return json.loads(output)
+
+
+def try_get_crate_manifest_path_from_mainfest_path(manifest_path: str, crate_name: str) -> str | None:
+    metadata = get_manifest_metadata(manifest_path)
+    crate_manifest_path = next((pkg["manifest_path"] for pkg in metadata["packages"] if pkg["name"] == crate_name), None)
+    return crate_manifest_path
+
+
+def find_crate_manifest_in_tree(tree: str, crate_name: str):
+    # in some cases Cargo.toml is not located at the top level, so we also look at subdirectories
+    manifest_paths = (os.path.join(tree, path) for path in glob.iglob("**/Cargo.toml", root_dir=tree, recursive=True))
+
+    for manifest_path in manifest_paths:
+        res = try_get_crate_manifest_path_from_mainfest_path(manifest_path, crate_name)
+        if res is not None:
+            return res
+
+    raise Exception(f"Couldn't find manifest for crate {crate_name} inside {tree}")
+
+
+def write_checksum_json(dir: str):
+    with open(os.path.join(dir, ".cargo-checksum.json"), "w") as f:
+        f.write('{"files":{},"package":null}')
+
+
+def create_vendor(vendor_staging_dir: str, out_dir: str):
+    lockfile_path = os.path.join(vendor_staging_dir, "Cargo.lock")
+    cargo_toml = load_file(lockfile_path)
+    os.makedirs(out_dir, exist_ok=True)
+    shutil.copy(lockfile_path, os.path.join(out_dir, "Cargo.lock"))
+
+    configLines = [
+        '[source.crates-io]',
+        'replace-with = "vendored-sources"',
+        '[source.vendored-sources]',
+        'directory = "cargo-vendor"',
+    ]
+
+    seen_keys = set()
+    for pkg in cargo_toml["package"]:
+        # ignore local dependenices
+        if "source" not in pkg.keys():
+            continue
+        source = pkg["source"]
+
+        if source.startswith("git+"):
+
+            source_info = parse_git_source(pkg["source"])
+            git_sha_rev = source_info["git_sha_rev"]
+
+            git_tree = os.path.join(vendor_staging_dir, "git", git_sha_rev)
+            crate_manifest_path = find_crate_manifest_in_tree(git_tree, pkg["name"])
+            crate_tree = os.path.dirname(crate_manifest_path)
+
+            dir_name = f"{pkg["name"]}-{pkg["version"]}"
+            crate_out_dir = os.path.join(out_dir, dir_name)
+            shutil.copytree(crate_tree, crate_out_dir)
+            os.chmod(crate_out_dir, 0o755)
+
+            with open(os.path.join(crate_out_dir, ".cargo-checksum.json"), "w") as f:
+                f.write('{"files":{},"package":null}')
```

Comment:
This could be replaced by a call to `write_checksum_json`

Reply:
This function was accidentally left behind